### PR TITLE
Continued changes for NNF fencing

### PR DESCRIPTION
**Pull request by NateThornton**
1. Updates the readme for

    - redfish (final and verified)
    - nnf (pending)

2. Adds my in-progress fence_nnf.py changes
3. Adds my notes.md file

Signed-off-by: Nate Thornton <nate.thornton@hpe.com>

### CODE REVIEW

File: docs/guides/ha-cluster/readme.md [hunk 1]
```diff
@@ -24,34 +24,42 @@ HPE hardware implements software known as the Hardware System Supervisor (HSS),
...
-
 The [Redfish fencing agent](https://github.com/ClusterLabs/fence-agents/tree/main/agents/redfish) from [ClusterLabs](https://github.com/ClusterLabs/fence-agents) should be used for Compute nodes in the cluster. Configure the agent with the following parameters:
 
 | Argument | Definition |
 | -------- | ---------- |
-| `--ip=[ADDRESS]` | The IP address or hostname of the compute node's HSS node controller |
-|`--systems-uri=[URI]` | The URI of the Systems object. Must be `/redfish/v1/Systems/Node0` |
-|`--ssl-insecure` | Instructs the use of an insecure SSL exchange |
+| `ip=[ADDRESS]` | The IP address or hostname of the HSS controller |
+| `port=[PORT]` | The Port of the HSS controller. Must be `80` |
```
Comment (bdevcich-hpe) [inline docs/guides/ha-cluster/readme.md:32]: should this be `port=80` then?

File: docs/guides/ha-cluster/readme.md [hunk 2]
```diff
@@ -24,34 +24,42 @@ HPE hardware implements software known as the Hardware System Supervisor (HSS),
...
+| `systems-uri=[URI]` | The URI of the Systems object. Must be `/redfish/v1/Systems/Node0` |
```
Comment (bdevcich-hpe) [inline docs/guides/ha-cluster/readme.md:33]: Same here

File: docs/guides/ha-cluster/readme.md [hunk 3]
```diff
@@ -24,34 +24,42 @@ HPE hardware implements software known as the Hardware System Supervisor (HSS),
...
+| `ssl-insecure=[BOOL]` | Instructs the use of an insecure SSL exchange. Must be `true` |
```
Comment (bdevcich-hpe) [inline docs/guides/ha-cluster/readme.md:34]: Same for these two?

File: docs/guides/ha-cluster/fence_nnf.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
+#!@PYTHON@
+
+# Fence agent for Near Node Flash devices
+
+import sys
+import logging
+import atexit
+
+sys.path.append("@FENCEAGENTSLIBDIR@")
```
Comment (bdevcich-hpe) [inline docs/guides/ha-cluster/fence_nnf.py:9]: I think we need a comment here. Does the user need to change this?

File: docs/guides/ha-cluster/fence_nnf.py [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
+from fencing import *
+from fencing import fail, fail_usage, run_delay
+
+try:
+    from kubernetes import client
+except ImportError:
+    logging.error("Couldn't import kubernetes.client - not found or not accessible")
```
Comment (bdevcich-hpe) [inline docs/guides/ha-cluster/fence_nnf.py:17]: Why handle this one specifically?

File: docs/guides/ha-cluster/fence_nnf.py [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+    all_opt["localconfig"] = {
+        "getopt" : "",
+        "longopt" : "localconfig",
+        "help" : "--localconfig                         Use the local kubernetes config",
+        "required" : "0",
+        "order" : 4
+    }
+
+def main():
+    atexit.register(atexit_handler)
```
Comment (bdevcich-hpe) [inline docs/guides/ha-cluster/fence_nnf.py:169]: Where is `atexit_handler`?

File: docs/guides/ha-cluster/fence_nnf.py [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+
+    try:
+        node = client.CoreV1Api(api_client).read_node(name)
+
+    except client.exceptions.ApiException as e:
+        logging.debug("Exception when reading node: %s %s", e, type(e))
+        if e.status == 401:
+            fail(EC_LOGIN_DENIED)
+        elif e.status == 404:
+            fail(EC_STATUS)
```
Comment (bdevcich-hpe) [inline docs/guides/ha-cluster/fence_nnf.py:34]: could use http.HTTPStatus for these codes

**Review by bdevcich-hpe — APPROVED** [review]

File: docs/guides/ha-cluster/fence_nnf.py (hunk 5 above)
Comment (NateThornton) [reply docs/guides/ha-cluster/fence_nnf.py:17]: Because it has to be manually installed using pip

File: docs/guides/ha-cluster/fence_nnf.py (hunk 4 above)
Comment (NateThornton) [reply docs/guides/ha-cluster/fence_nnf.py:9]: Sorry, should have made it more obvious in the PR. This is just a place to stuff my changes so I have a copy if my machine were to crash. The final place will be in the ClusterLabs repo (or a fork), in which case configure/make will handle the string substitution.

File: docs/guides/ha-cluster/fence_nnf.py (hunk 6 above)
Comment (NateThornton) [reply docs/guides/ha-cluster/fence_nnf.py:169]: It's in fencing.py

**Pull request merged**